src/badge/badge.js

```diff
@@ -1,20 +1,31 @@
 import React from 'react';
-import { Text, View, StyleSheet } from 'react-native';
+import { Text, View, StyleSheet, TouchableOpacity } from 'react-native';
 
 let styles = {};
 
 const Badge = props => {
   const { badge } = props;
 
+  const {
+    badgeContainerStyle,
+    badgeTextStyle,
+    onPress,
+    element,
+    value
+  } = props;
+
   if (!badge) throw Error('badge prop is required');
 
-  if (badge.element) return badge.element;
+  if (element) return badge.element;
+
+  const Component = onPress !== undefined ? TouchableOpacity : View;
 
   return (
-    <View style={[ styles.badge, badge.badgeContainerStyle ]}>
-      <Text style={[ styles.text, badge.badgeTextStyle ]}>{badge.value}</Text>
-    </View>
+    <Component style={[ styles.badge, badgeContainerStyle ]} onPress={onPress}>
```

Comment:
You should pass `{...attributes}` this let's user pass all available props for any component they use.